frame/contracts/src/tests.rs

```diff
@@ -580,6 +580,43 @@ fn storage_max_value_limit() {
 	});
 }
 
+#[test]
+fn call_code() {
+	let (caller_wasm, caller_code_hash) = compile_module::<Test>("call_code").unwrap();
+	let (callee_wasm, callee_code_hash) = compile_module::<Test>("call_code_lib").unwrap();
+	let caller_addr = Contracts::contract_address(&ALICE, &caller_code_hash, &[]);
+
+	ExtBuilder::default().existential_deposit(500).build().execute_with(|| {
+		let _ = Balances::deposit_creating(&ALICE, 1_000_000);
+
+		// Instantiate the 'caller'
+		assert_ok!(Contracts::instantiate_with_code(
+			Origin::signed(ALICE),
+			300_000,
+			GAS_LIMIT,
+			None,
+			caller_wasm,
+			vec![],
+			vec![],
+		));
+		// Only deploy 'callee' code
+		assert_ok!(Contracts::upload_code(
+			Origin::signed(ALICE),
+			callee_wasm,
+			Some(codec::Compact(100_000)),
+		));
+
+		assert_ok!(Contracts::call(
```

Comment:
Is that possible to check that we really executed the library's code? Maybe we can set something to storage and get the result in the test

Reply:
we assert instruction in the caller contract that checks that value has been changed by the callee contract code